### ISSUE TITLE
Type icons

### ISSUE THREAD

**Issue by GameModerator** [msikma/pokesprite#84]
Hey, would it be possible to extract the category type icons such as physical, special and status too? I found the type-type icons but so far no luck with the one I mentioned. Thanks if you can!

**Comment by msikma** [comment]
Hi there, generally icons like this are not included in this project because they're UI elements. But I suppose you can make the argument that these would still be useful to have because they can be reused in many different contexts.

The difficulty is that UI elements usually need to fit into a UI. Which is why usually you'd make custom ones to fit into the specific design for your application. Like say, these ones by Smogon are made to fit their design:

<img width="534" alt="Screenshot 2020-08-01 at 15 21 45" src="https://user-images.githubusercontent.com/1507745/89102543-bc2c4200-d40a-11ea-9057-cd305375aeed.png">

If we include icons for physical/special/status, would that be just the star and circle icons or should they include the background? Should they be ripped from Sword/Shield? In that case they'll be slanted to the right, since that's what makes sense for that specific UI design:

<img width="179" alt="Screenshot 2020-08-01 at 15 26 06" src="https://user-images.githubusercontent.com/1507745/89102617-58564900-d40b-11ea-8dfa-88c378b9752f.png">

I'm not saying these icons should not be included in the project, but we need to think a bit about what would be most useful for users, and how they'll be used.

**Comment by GameModerator** [comment]
> If we include icons for physical/special/status, would that be just the star and circle icons
Yup
> Should they include the background?
Nope
> Should they be ripped from Sword/Shield?
Yep

Setting a background wouldn't be hard, especially with css.

Well if it's not too much trouble, I do need those icons, and the Dynamax ones too if you can rip those too. I have neither the tools nor expertise to do them myself and so far.. google hasn't been much help either.

**Comment by msikma** [comment]
Alright, fair enough. I'm actually looking into the move type icons (physical/special/status), but I haven't been able to find nice versions of these. So I'm thinking I'll just make screenshots from in-game and mask them out. Aside from that I think I'll make a vector custom one since the in-game ones are rather small.

I'm also looking into type icons, both white transparent ones and colored ones.

My aim is to provide a couple of different options to fit multiple use cases.

**Comment by koenigderluegner** [comment]
This might help for type icons:
https://www.spriters-resource.com/nintendo_switch/pokemonswordshield/sheet/123715/

I'm not sure if that fits to this topic but since we got the Gmax icon, would it make sense to get other pokemon related icons like shiny stars or origin, too?

**Comment by msikma** [comment]
> This might help for type icons:
> https://www.spriters-resource.com/nintendo_switch/pokemonswordshield/sheet/123715/

I've looked into using this image, but I've actually gotten them from Bulbapedia instead because they're higher resolution there. I'm not sure how they were ripped. I'm adding the type icons verbatim (with color) and masked out versions (just pure white on transparent).

<img width="876" alt="image" src="https://user-images.githubusercontent.com/1507745/89692156-337e3c00-d90b-11ea-9884-859aaee7b2cf.png">

Also, the reason why I'm generally a bit reluctant to add UI elements is because each one is going to have limited use cases. Things like box sprites are immutable, but whether e.g. a shiny star works is very dependent on circumstances. For example, I'm planning on making the transparent type icons white so you can put a background behind them or use it as mask, but eventually someone is going to say "my interface is white, can we have black versions of these?" There are a lot more variables.

That's also why I asked those questions earlier. It's not that I'm not open to adding more UI elements, but I'm just trying to figure out what would work the best for most people.

**Comment by koenigderluegner** [comment]
That makes totally sense. I'm not a Photoshop expert (or whatever you use) but did you consider making them as SVG when you redraw them? This would solve all scaling and coloring problems

**Comment by msikma** [comment]
I'm not actually redrawing the images, all I'm doing is masking them out of the background so that the background color becomes transparent. Making SVG versions would be a lot of work and I'm not really cut out to do it anyway.

In any case, I've added the type icons and transparent type logos to master: https://msikma.github.io/pokesprite/overview/misc.html#types

The move type icons will be coming later.

**Comment by GameModerator** [comment]
> "my interface is white, can we have black versions of these?"

`filter: invert(1);`

**Comment by msikma** [comment]
> filter: invert(1);

I'm sorry but you're not getting my point. My point is there are a thousand different potential situations, such as where you *can't use CSS filters*.

That's why I need to be selective, and why I won't always be able to simply accommodate someone who comes around and says "put this in the project" and "I need it exactly like this".

**Comment by msikma** [comment]
On another note, \@edcrfv in the PP Discord tipped me off to this:

<img width="1057" alt="image" src="https://user-images.githubusercontent.com/1507745/89885659-b03f3d80-dbcb-11ea-9e61-7b68b4647f74.png">

Official SVG versions of the type icons. This is what was used to create the Bulbapedia images (if only they'd mention these things in the image descriptions...)

They even knew the font used for the type names:

* https://fontworks.co.jp/fontsearch/udkakugoc80pro-b/?word=NORMAL
* https://fontworks.co.jp/fontsearch/rodinntlgpro-b/?word=%E3%83%8E%E3%83%BC%E3%83%9E%E3%83%AB

**Comment by GameModerator** [comment]
> > filter: invert(1);
> 
> But there are a thousand different potential situations, such as where you can't use CSS filters. This is not an issue with all of the non-UI sprites because they are used verbatim.

That was a joke lmao. I know there's a lot of ways, that was just one of them. 

What happened to the category type icons if I may ask?

**Comment by msikma** [comment]
> What happened to the category type icons if I may ask?

Coming soon 🙂

**Comment by msikma** [comment]
I've made a few changes.

* New png renders for the type icons from the original SVG files
* The type logos have been replaced with 128x128 versions (just like the icons) of the inverted SVG versions, and they're now in their proper color rather than white since this is more flexible
* I've added the physical/special/status icons

<img width="1301" alt="Screenshot 2020-08-15 at 19 49 31" src="https://user-images.githubusercontent.com/1507745/90318436-be28f180-df30-11ea-8d51-8435133a05f2.png">

The larger versions are from Pokémon Home. Then there's the version from Gen 8, which I've had to clean up a little since I'm only able to get screenshots using the Switch's regular screenshot method which makes jpg files. But I think they look OK now. They're black on transparent.

The docs have been updated and I've made this version 2.2.3.

Let me know if this works for your purposes.
